### PR TITLE
docs: Add example on how to integrate servicefoundry with ci/cd.

### PR DESCRIPTION
**Pull request by akshatdalton**

### CODE REVIEW

File: ci-cd-automation/servicefoundry.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      type: build
+      build_source:
+        type: local
+      build_spec:
+        type: tfy-python-buildpack
+        python_version: 3.9
+        requirements_path: requirements.txt
+        command: uvicorn main:app --port 4000 --host 0.0.0.0
+    ports:
+      - expose: True
```
Comment (chiragjn) [inline ci-cd-automation/servicefoundry.yaml:15]: this can be removed
```suggestion
```

File: ci-cd-automation/servicefoundry.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  - name: ci-cd-example
+    type: service
+    image:
+      type: build
+      build_source:
+        type: local
+      build_spec:
+        type: tfy-python-buildpack
+        python_version: 3.9
+        requirements_path: requirements.txt
```
Comment (chiragjn) [inline ci-cd-automation/servicefoundry.yaml:12]: this can be removed
```suggestion
```

File: ci-cd-automation/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+fastapi
+uvicorn[standard]
```
Comment (chiragjn) [inline ci-cd-automation/requirements.txt:2]: pin these to specific version or ranges

File: ci-cd-automation/main.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from fastapi import FastAPI
```
Comment (chiragjn) [inline ci-cd-automation/main.py:1]: Maybe move this whole example under
`deployment/`

File: .github/workflows/ci-cd-automation.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+name: Servicefoundry CI/CD Automation Example
```
Comment (chiragjn) [inline .github/workflows/ci-cd-automation.yml:1]: Shouldn't we place this at 

```
deployment/ci-cd-automation/.github/workflows/ci-cd-automation.yml
```

And then we explain in README that in an actual repository `.github` should be at the root of the repo with a tree structure

File: .github/workflows/ci-cd-automation.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      WORKSPACE_FQN: tfy-dev-cluster:ci-cd-examples
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up python
+        uses: actions/setup-python@v3
+        with:
+          python-version: 3.9
+
+      - name: Install servicefoundry-cli library
+        run: pip install servicefoundry
```
Comment (chiragjn) [inline .github/workflows/ci-cd-automation.yml:23]: pin to a specific version range
```suggestion
        run: pip install "servicefoundry>=0.2.14,<0.3.0"
```

File: .github/workflows/ci-cd-automation.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+      - name: Deploy on servicefoundry server
+        working-directory: ./ci-cd-automation
+        run: sfy deploy --file servicefoundry.yaml --workspace-fqn $WORKSPACE_FQN
```
Comment (chiragjn) [inline .github/workflows/ci-cd-automation.yml:27]: where will `$WORKSPACE_FQN` come from 🤔 ?

File: .github/workflows/ci-cd-automation.yml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+on:
+  push:
+    paths:
+      - 'ci-cd-automation/*'
+
+jobs:
+  production_deploy:
+    runs-on: ubuntu-latest
+    env:
+      TFY_HOST: https://app.develop.truefoundry.tech
```
Comment (chiragjn) [inline .github/workflows/ci-cd-automation.yml:12]: Not exactly sure about this one, I think we should keep this to production
```suggestion
      TFY_HOST: https://app.truefoundry.com/
```

File: .github/workflows/ci-cd-automation.yml [hunk 9]
```diff
@@ -0,0 +1,27 @@
+name: Servicefoundry CI/CD Automation Example
+
+on:
+  push:
+    paths:
+      - 'ci-cd-automation/*'
```
Comment (chiragjn) [inline .github/workflows/ci-cd-automation.yml:6]: I think there should be branch scoping too, i.e. only deploy if there is a push to `main` branch in this folder

Comment (chiragjn) [reply .github/workflows/ci-cd-automation.yml:6]: Also change this accordingly if you move around the folder

File: .github/workflows/ci-cd-automation.yml [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+jobs:
+  production_deploy:
+    runs-on: ubuntu-latest
+    env:
+      TFY_HOST: https://app.develop.truefoundry.tech
+      TFY_API_KEY: ${{ secrets.TFY_API_KEY }}
```
Comment (chiragjn) [inline .github/workflows/ci-cd-automation.yml:13]: We should add a readme linking to our docs page
I am hoping we explain/link how to create a Github Secret

File: ci-cd-automation/servicefoundry.yaml (hunk 1 above)
Comment (akshatdalton) [reply ci-cd-automation/servicefoundry.yaml:15]: is that a default setting?

File: .github/workflows/ci-cd-automation.yml (hunk 8 above)
Comment (akshatdalton) [reply .github/workflows/ci-cd-automation.yml:12]: yup, I don't know whose account to use to deploy there

File: ci-cd-automation/servicefoundry.yaml (hunk 1 above)
Comment (debajyoti-truefoundry) [reply ci-cd-automation/servicefoundry.yaml:15]: yes

File: .github/workflows/ci-cd-automation.yml (hunk 10 above)
Comment (debajyoti-truefoundry) [reply .github/workflows/ci-cd-automation.yml:13]: There should be a screenshot on the corresponding docs page.

File: ci-cd-automation/servicefoundry.yaml (hunk 1 above)
Comment (chiragjn) [reply ci-cd-automation/servicefoundry.yaml:15]: yes, I think we even removed it completely from the spec

File: .github/workflows/ci-cd-automation.yml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    branches:
+      - main
+    paths:
+      - 'deployment/ci-cd-automation/*'
+
+jobs:
+  production_deploy:
+    runs-on: ubuntu-latest
+    env:
+      TFY_HOST: https://app.develop.truefoundry.tech
```
Comment (chiragjn) [inline .github/workflows/ci-cd-automation.yml:14]: Let's update this to prod later

**Review by chiragjn — APPROVED** [review]

**Pull request merged**